.github/workflows/ci.yml

```diff
@@ -251,6 +340,7 @@ jobs:
       matrix:
         include: ${{ fromJson(needs.setup.outputs.test_matrix) }}
     steps:
+      - uses: actions/checkout@v4
```

Comment:
It's not clear to me why you are checking out the repo in this job; the Boost test runner does not require anything other than the test binary.

Reply:
Indeed, it's not necessary. Good catch. The fix will be included in the next commit

Reply:
It is added again due to the composite action